gdnative-core/src/core_types/geom/transform.rs

```diff
@@ -157,6 +186,123 @@ impl Transform {
         }
     }
 
+    /// In-place scales basis and origin of the transform by the given scale 
+    /// factor, using matrix multiplication.
+    #[inline]
+    pub fn scale(&mut self, scale:Vector3) {
+        self.scale_basis(scale);
+        self.origin *= scale;
+    }
+
+    /// Scales basis and origin of the transform by the given scale factor, 
+    /// using matrix multiplication.
+    #[inline]
+    pub fn scaled(&self, scale:Vector3) -> Self {
+        let mut copy = *self;
+        copy.scale(scale);
+        copy
+    }
+
+    /// In-place scales basis of the transform by the given scale factor
+    #[inline]
+    fn scale_basis(&mut self, scale:Vector3) {
+        self.basis.scale_self(scale)
+    }
+
+    /// In-place translates the transform by the given offset, relative to 
+    /// the transform's basis vectors.
+    #[inline]
+    pub fn translate_float(&mut self, tx:f32, ty:f32, tz:f32) {
+        self.translate(Vector3::new(tx, ty, tz))
+    }
```

Comment:
We probably don't need this one, as constructing a vector is quite simple. Otherwise, we would need such convenience overloads all over the API 😉

(same for rotate/scale)